microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/jaxrs/java/JaxRsCodeLensParticipant.java

```diff
@@ -58,30 +70,44 @@
 	private static final int PING_TIMEOUT = 2000;
 
 	@Override
-	public boolean isAdaptedForCodeLens(JavaCodeLensContext context, IProgressMonitor monitor) throws CoreException {
+	public boolean isAdaptedForCodeLens(JavaCodeLensContext context,
+			IProgressMonitor monitor) throws CoreException {
 		MicroProfileJavaCodeLensParams params = context.getParams();
 		if (!params.isUrlCodeLensEnabled()) {
 			return false;
 		}
 		// Collection of URL codeLens is done only if JAX-RS is on the classpath
 		IJavaProject javaProject = context.getJavaProject();
-		return JDTTypeUtils.findType(javaProject, JAVAX_WS_RS_PATH_ANNOTATION) != null;
+
+		return JDTTypeUtils.findType(javaProject,
+				JAVAX_WS_RS_PATH_ANNOTATION) != null;
 	}
 
 	@Override
-	public List<CodeLens> collectCodeLens(JavaCodeLensContext context, IProgressMonitor monitor) throws CoreException {
+	public List<CodeLens> collectCodeLens(JavaCodeLensContext context,
+			IProgressMonitor monitor) throws CoreException {
 		ITypeRoot typeRoot = context.getTypeRoot();
 		IJavaElement[] elements = typeRoot.getChildren();
 		JaxRsContext jaxrsContext = JaxRsContext.getJaxRsContext(context);
 		IJDTUtils utils = context.getUtils();
 		MicroProfileJavaCodeLensParams params = context.getParams();
 		List<CodeLens> lenses = new ArrayList<>();
-		collectURLCodeLenses(elements, null, lenses, params, jaxrsContext, utils, monitor);
+
+		IJavaProject javaProject = context.getJavaProject();

```

Comment:
To be consistent with Quarkus codelens at https://github.com/redhat-developer/quarkus-ls/blob/209e7463ecbe7d8c58f81cfe20137019c075049b/quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/jaxrs/java/QuarkusJaxRsCodeLensParticipant.java#L57 move the code which retrieve the AppplicationPath in the beginCodelens liekhttps://github.com/redhat-developer/quarkus-ls/blob/209e7463ecbe7d8c58f81cfe20137019c075049b/quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/jaxrs/java/QuarkusJaxRsCodeLensParticipant.java#L42

Reply:
I believe isn't a feature that interacts with the QuarkusJaxRsContext, since the retrieval of the `@ApplicationPath` value comes from the java project (i.e. same way the `@Path` annotation is retrieved). By moving this, I need to store this in the JaxRsContext on quarkus-ls, is this alright?

Reply:
I think you don't need this PR. The main idea is that MP contribute to JaxRsContext with application path and that's it.

Reply:
I was under the impression that we needed to move this as per https://github.com/eclipse/lsp4mp/pull/195#discussion_r725009092, correct?

Reply:
> I have the impresion that ApplicationPath value should update JaxRSContext#setRootPath, no?

rootPath comes from application.properties (quarkus.http.rooth.path), where as ApplicationPath comes from the `@ApplicationPath` annotation